src/Umbraco.Core/Persistence/Repositories/EntityRepository.cs

```diff
@@ -331,24 +331,49 @@ public virtual IEnumerable<IUmbracoEntity> GetByQuery(IQuery<IUmbracoEntity> que
             return list;
         }
 
+        /// <summary>
+        /// Gets entities by query.
+        /// Note that this will also fetch all property data for media items, which can cause performance problems
+        /// when used without paging, in sites with large amounts of data in cmsPropertyData.
+        /// </summary>
+        /// <param name="query"></param>
+        /// <param name="objectTypeId"></param>
+        /// <returns></returns>
         public virtual IEnumerable<IUmbracoEntity> GetByQuery(IQuery<IUmbracoEntity> query, Guid objectTypeId)
         {
+            return GetByQueryInternal(query, objectTypeId, true);
+        }
 
-            bool isContent = objectTypeId == Constants.ObjectTypes.DocumentGuid || objectTypeId == Constants.ObjectTypes.DocumentBlueprintGuid;
-            bool isMedia = objectTypeId == Constants.ObjectTypes.MediaGuid;
+        /// <summary>
+        /// Gets entities by query without fetching property data.
+        /// This is supposed to be internal and can be used when getting all entities without paging, without causing
+        /// performance issues.
+        /// </summary>
+        /// <param name="query"></param>
+        /// <param name="objectTypeId"></param>
+        /// <returns></returns>
+        internal IEnumerable<IUmbracoEntity> GetByQueryWithoutPropertyData(IQuery<IUmbracoEntity> query, Guid objectTypeId)

```

Comment:
There's no reason to have the objectTypeId param here, this methods should just be called GetMediaByQueryWithoutPropertyData